.github/workflows/full_kubeflow_integration_test.yaml

```diff
@@ -0,0 +1,361 @@
+    branches:
+      - master
+
+jobs:
+  build:
+    name: Full E2E Integration Test
+    runs-on:
+      labels: ubuntu-latest-16-cores
+    env:
+      KIND_CLUSTER_NAME: kubeflow
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.10'
+        
+    - name: Install Python dependencies
+      run: |
+        pip install pytest kubernetes kfp==2.11.0 kserve pytest-timeout pyyaml requests
+
+    - name: Install KinD, Create KinD cluster and Install kustomize
+      run: ./tests/gh-actions/install_KinD_create_KinD_cluster_install_kustomize.sh
+
+    - name: Install kubectl
+      run: ./tests/gh-actions/install_kubectl.sh
+
+    - name: Create kubeflow namespace
+      run: kustomize build common/kubeflow-namespace/base | kubectl apply -f -
+
+    - name: Install Cert Manager
+      run: ./tests/gh-actions/install_cert_manager.sh
+
+    - name: Install Istio
+      run: ./tests/gh-actions/install_istio.sh
+
+    - name: Install oauth2-proxy
+      run: ./tests/gh-actions/install_oauth2-proxy.sh
+
+    - name: Deploy Full Kubeflow Stack
+      run: |
+        # Apply with retry logic using exact command from README
+        echo "Installing full Kubeflow stack from example/kustomization.yaml"
+        while ! kustomize build example | kubectl apply --server-side --force-conflicts -f -; do 
+          echo "Retrying to apply resources"; 
+          sleep 20; 
+        done
+        
+        # Check deployment status
+        kubectl get deployments --all-namespaces
+        kubectl get pods --all-namespaces
+
+    - name: Install kubeflow-istio-resources
+      run: kustomize build common/istio-1-24/kubeflow-istio-resources/base | kubectl apply -f -
+
+    - name: Wait for Critical Services
+      run: |
+        echo "Waiting for all deployments to be ready..."
+        kubectl wait --for=condition=available --timeout=1200s deployment --all -n cert-manager || true
+        kubectl wait --for=condition=available --timeout=1200s deployment --all -n istio-system || true
+        kubectl wait --for=condition=available --timeout=1200s deployment --all -n auth || true
+        kubectl wait --for=condition=available --timeout=1800s deployment --all -n kubeflow || true
+        
+        # Wait for specific critical components to be ready
+        kubectl -n kubeflow wait --for=condition=Ready pods -l kustomize.component=profiles --timeout 180s || true
+        kubectl -n kubeflow wait --for=condition=Ready pods -l app=centraldashboard --timeout 180s || true
+        kubectl -n kubeflow wait --for=condition=Ready pods -l app=ml-pipeline --timeout 180s || true
+        kubectl -n istio-system wait --for=condition=Ready pods --all --timeout 180s || true
+        kubectl -n cert-manager wait --for=condition=Ready pods --all --timeout 180s || true
+        kubectl -n auth wait --for=condition=Ready pods --all --timeout 180s || true
```

Comment:
I think this is already covered in the installation scripts under /tests and can be removed. If not please add it to the install scripts.